runtime/src/accounts_db.rs

```diff
@@ -1515,7 +1518,7 @@ impl AccountsDb {
 
         let mut collect_delta_keys = Measure::start("key_create");
         let max_slot = max_clean_root.unwrap_or_else(|| self.accounts_index.max_root());
-        let (delta_keys, _max_slot) = self.collect_uncleaned_pubkeys_to_slot(max_slot);
```

Comment:
Since `max_slot` is never used, I figured I'd remove it from here, and therefore remove it from the functions. I'm assuming that's OK, yes?

Reply:
@brooksprumo yeah, it looks just an oversight from quick look. I think you can check `git log` of relevant lines to make informed verdict for that. :) And could you write short findings from the historian work here?

Reply:
OK, looks like the `max_slot` was used in the original commit `ca7914aa3d73f17217de498a4e5dcec3f0b9ce0d`, but was then reverted the next day in commit `ad2e10e17b521d6a880a5e4958787d1cbd55247a`.

Two days after that, in commit `0d32a0e0f44f53ba13189c10ee4b85da2cc8ca68`, `collect_uncleaned_pubkeys_to_slot()` was re-added via a re-revert (un-revert?), but the use of the `max_slot` return parameter was removed. Inside `construct_candidate_clean_keys()`, when `collect_uncleaned_pubkeys_to_slot()` is called, the `max_slot` was always ignored, so my guess is the re-revert should've removed `max_slot` here, but wasn't.

Reply:
i see. perfect answer. thanks for the write up.